app/models/goobi_xml_import.rb

```diff
@@ -3,15 +3,30 @@
 class GoobiXmlImport < ApplicationRecord
   attr_reader :file
   before_create :file
-  validate :check_for_oid
+  validate :oid, :valid_mets?, :all_images_present?
 
   def file=(value)
     @file = value
     self[:goobi_xml] = value.read
   end
 
-  def check_for_oid
+  def oid
     mets_doc = MetsDocument.new(@file)
-    errors.add(:file, 'must contain an oid') if mets_doc.oid.nil? || mets_doc.oid.empty?
+    return errors.add(:file, 'must contain an oid') if mets_doc.oid.nil? || mets_doc.oid.empty?
```

Comment:
validation and accessing should be seperate. this method currently has 2 concerns.  I would suggest making def oid return nil and then having a method that validates seperately